src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VisualBasic/VisualBasicVSImports.cs

```diff
@@ -12,6 +12,7 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation.VisualBasic
 {
     [Export(typeof(Imports))]
     [Export(typeof(ImportsEvents))]
+    [Export(typeof(VisualBasicVSImports))]
     [AppliesTo(ProjectCapability.VisualBasic)]
     [Order(Order.Default)]
     internal class VisualBasicVSImports : ConnectionPointContainer,
```

Comment:
I think we should have a similar relationship between this and VisualBasicNamespaceImportsList like we do with OSVSReferences and ReferenceHostBridge. It doesn't feel like VisualBasicNamespaceImportsList should know anything about firing VS events, and VisualBasicVSImports waits for results from the HostBridge.

Reply:
Or maybe jsut combine the two like VsPersistConnectedServices? Their relationship is a circular reference.

Reply:
Oh my bad, that's exactly what ReferenceHostBridge does. Ugh - circular dependencies are the pits.